ci-operator/step-registry/telco5g/cnf/tests/telco5g-cnf-tests-commands.sh

```diff
@@ -294,100 +213,36 @@ if [[ "$CNF_BRANCH" == *"4.13"* ]] || [[ "$CNF_BRANCH" == *"master"* ]]; then
 
-cd -
+skip_tests=$(get_skip_tests)
 
-# check tests results
-exit_code=0
-err_msg=""
-rm -f summary.txt
-for feature in ${FEATURES}; do
-    log_file="${ARTIFACT_DIR}/deploy_and_test_${feature}.log"
-    if [ ! -f "${log_file}" ]; then
-        err_msg="${err_msg}\n[ERROR]: Failed to test ${feature}"
-        exit_code=1
-    else
-        # this sed removes coloring from log
-        sed -i "s,\x1B\[[0-9;]*[a-zA-Z],,g" "${log_file}"
-        # check if actually reached the stage of integration tests
-        if [ ! "$(grep --text "Running Suite: CNF Features e2e integration tests" "${log_file}")" ]; then
-            err_msg="${err_msg}\n[ERROR]: ${feature} testing didn't reach integration tests"
-            exit_code=1
-        else
-            # get integration tests results from the full log file
-            rm -f temp_summary.log
-            stage_first_line=$(grep --text -n -m 1 "Running Suite: CNF Features e2e integration tests" "${log_file}" | cut -f1 -d:)
-            tail --lines=+"${stage_first_line}" "${log_file}" >> temp_summary.log
-            SUCCESS=$(grep --text 'Ran.*of.*Specs in.*seconds' temp_summary.log -A 1 | head -2  | tail -1)
-            if [ -z "${SUCCESS}" ]; then
-                err_msg="${err_msg}\n[ERROR]: ${feature} tests failed"
-                exit_code=1
-            elif [[ "${SUCCESS}" != *"SUCCESS"* ]]; then
-                SUMMARY=$(grep --text 'Summarizing.*Failure*' temp_summary.log)
-                if [ -n "${SUMMARY}" ]; then
-                    cat temp_summary.log | sed -n '/Summarizing.*Failure*/,/Ran.*of.*Specs in.*seconds.*/p' | sed -e '$d' >> summary.txt
-                fi
-                err_msg="${err_msg}\n[ERROR]: ${feature} tests failed"
-                exit_code=1
-            fi
-        fi
+worker_nodes=$(oc get nodes --selector='node-role.kubernetes.io/worker' \
+--selector='!node-role.kubernetes.io/master' -o name)
+if [ -z "${worker_nodes}" ]; then
+    echo "[ERROR]: No worker nodes found in cluster"
+    exit 1
+fi
+# get BM workers for testing
+test_nodes=""
+for node in ${worker_nodes}; do
+    if is_bm_node "${node}"; then
+        test_nodes="${test_nodes} ${node}"
     fi
 done
 
-echo -e "${err_msg}\n"
-if [ -f "summary.txt" ]; then
-    cat summary.txt
+if [ -z "${test_nodes}" ]; then
+    echo "[ERROR]: No BM worker nodes found in cluster"
+    exit 1
 fi
 
-# Create a HTML report
-for feature in ${FEATURES}; do
-    xml_f="${ARTIFACT_DIR}/${feature}/cnftests-junit.xml"
-    if [[ -f $xml_f ]]; then
-        cp $xml_f ${ARTIFACT_DIR}/cnftests-junit_${feature}.xml
-    fi
-    xml_v="${ARTIFACT_DIR}/${feature}/validation_junit.xml"
-    if [[ -f $xml_v ]]; then
-        cp $xml_v ${ARTIFACT_DIR}/validation_junit_${feature}.xml
-    fi
-    xml_s="${ARTIFACT_DIR}/${feature}/setup_junit.xml"
-    if [[ -f $xml_s ]]; then
-        cp $xml_s ${ARTIFACT_DIR}/setup_junit_${feature}.xml
-    fi
-done
+export CNF_NODES="${test_nodes}"
```

Comment:
if you are using all nodes, no need to set it

Reply:
oh i thought you removed it, i didn't see it in the artifacts

Reply:
I tried w/o it and w/ it, not sure if we need it though. Trying to get the answer :)